packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -212,6 +212,20 @@ export default function TypographyPanel( {
 	const mergedFontSizes = getMergedFontSizes( settings );
 
 	const fontSize = decodeValue( inheritedValue?.typography?.fontSize );
+
+	// Extract the slug from the CSS custom property if it exists
+	const currentFontSizeSlug = ( () => {
+		const rawValue = inheritedValue?.typography?.fontSize;
+		if (
+			rawValue &&
+			typeof rawValue === 'string' &&
+			rawValue.startsWith( 'var:preset|font-size|' )
+		) {
+			return rawValue.replace( 'var:preset|font-size|', '' );
+		}
+		return undefined;
+	} )();
```

Comment:
Any reason for the self-executing function?

This would also work the same way:

```js
	const currentFontSizeSlug =
		inheritedValue?.typography?.fontSize?.startsWith(
			'var:preset|font-size|'
		)
			? inheritedValue.typography.fontSize.replace(
					'var:preset|font-size|',
					''
			  )
			: undefined;
```

Reply:
> Any reason for the self-executing function?

Maybe only in that the current code guards against non-string types. 

I believe the `typography.fontSize` value must be a string so I don't see a lot of difference here, even in terms of performance.